dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -439,6 +439,7 @@ private LambdaEvent createBasicEvent(String repository, String gitReference, Str
 
                 Workflow workflow = createOrGetWorkflow(BioWorkflow.class, repository, user, workflowName, subclass, gitHubSourceCodeRepo);
                 workflow = addDockstoreYmlVersionToWorkflow(repository, gitReference, dockstoreYml, gitHubSourceCodeRepo, workflow);
+                workflow.updateLastModified();
```

Comment:
This `updateLastModified` implementation loads all versions. You had changed `addDockstoreYmlVersionToWorkflow` to not load all versions, so this will have a big performance impact.

Since `addDockstoreYmlVersionToWorkflow` only adds or modifies one version, can you instead just get the version's lastModified there, compare it to the workflow's, and update the workflow last modified if the version is newer? I forget if the lastModified is the Dockstore processing time or the Git commit time -- if it's the Git commit time, then it is possible for it to be older than the workflow's lastModified, confusingly.